### PR TITLE
Separate HepMC2 and HepMC3 build recipes; Python3 support for HepMC3

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 1]
```diff
@@ -28,24 +28,34 @@ class Hepmc(CMakePackage):
 
     variant('python', default=False, description='Enable Python bindings')
     variant('rootio', default=False, description='Enable ROOT I/O')
+    variant('interfaces', default=False, description='Install interfaces for some Monte-Carlo Event Gens')
 
     depends_on('cmake@2.8.9:', type='build')
     # FIXME: Officially supports Python3, but the build system doesn't find it
-    depends_on('python@:2.99.99', when='+python')
+    #depends_on('python@:2.99.99', when='+python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: Why is this commented out?

Comment (iarspider) [reply var/spack/repos/builtin/packages/hepmc/package.py]: I have a few corrections in local repo, will push soon.

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 2]
```diff
@@ -28,24 +28,47 @@ class Hepmc(CMakePackage):
...
+    def cmake_args(self):
+        return ['-Dmomentum:STRING=GEV', '-Dlength:STRING=MM']
+
+    @when('@3:')
     def cmake_args(self):
         spec = self.spec
-        return [
+        py_ver = spec['python'].version.up_to(2)
+        py_sitepkg = '{0}/lib/python{1}/site-packages'.format(
+            spec.prefix, py_ver)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: You should be able to do something like:
```python
py_sitepkg = join_path(self.prefix, site_packages_dir)
```
Also, remember to move this logic into the `if '+python' in self.spec:` below, as `python` is not always a dependency.
This variable is set by the `python` package.

Comment (iarspider) [reply var/spack/repos/builtin/packages/hepmc/package.py]: @adamjstewart thanks for the hints!

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 3]
```diff
@@ -28,24 +28,46 @@ class Hepmc(CMakePackage):
...
     variant('rootio', default=False, description='Enable ROOT I/O')
+    variant('interfaces', default=False, description='Install interfaces for some Monte-Carlo Event Gens')
 
     depends_on('cmake@2.8.9:', type='build')
-    # FIXME: Officially supports Python3, but the build system doesn't find it
-    depends_on('python@:2.99.99', when='+python')
+    depends_on('python', when='+python')
     depends_on('root', when='+rootio')
 
     conflicts('+python', when='@:3.1.99')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]:
```suggestion
    conflicts('+python', when='@:3.1')
```

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 4]
```diff
@@ -28,24 +28,46 @@ class Hepmc(CMakePackage):
...
     conflicts('+rootio', when='@:2.99.99')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]:
```suggestion
    conflicts('+rootio', when='@:2')
```

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 5]
```diff
@@ -28,24 +28,46 @@ class Hepmc(CMakePackage):
...
+    conflicts('+interfaces', when='@:2.99.99')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]:
```suggestion
    conflicts('+interfaces', when='@:2')
```

**Pull request merged**